### PR TITLE
[#185] Rename some update mechanism types

### PR DESCRIPTION
**Pull request by ruhatch**
This renames some of the main types in the update mechanism:
- `BlockVersion` -> `ProtocolVersion`
- `BlockVersionData` -> `ProtocolParameters`
- `BlockVersionModifier` -> `ProtocolParameterUpdate`
- `BlockVersionState` -> `ProtocolVersionState` (this one might change later)

### CODE REVIEW

File: src/Cardano/Chain/Block/ExtraHeaderData.hs [hunk 1]
```diff
@@ -21,15 +21,15 @@ import qualified Formatting.Buildable as B
...
 import Cardano.Chain.Update.SoftwareVersion
   (SoftwareVersion, SoftwareVersionError, checkSoftwareVersion)
 import Cardano.Crypto (Hash)
 
 
 -- | Represents block header extra data
 data ExtraHeaderData = ExtraHeaderData
-  { ehdBlockVersion    :: !BlockVersion
+  { ehdProtocolVersion :: !ProtocolVersion
   -- ^ Version of block
```
Comment (dnadales) [inline src/Cardano/Chain/Block/ExtraHeaderData.hs]: Do we need to adapt the comment as well?

File: src/Cardano/Chain/Genesis/Generate.hs [hunk 2]
```diff
@@ -214,14 +214,14 @@ generateGenesisData startTime genesisSpec = do
...
     genesisData = GenesisData
-      { gdBootStakeholders = bootStakeholders
-      , gdHeavyDelegation  = genesisDlg
-      , gdStartTime        = startTime
-      , gdNonAvvmBalances  = nonAvvmDistr
-      , gdBlockVersionData = gsBlockVersionData genesisSpec
-      , gdK                = gsK genesisSpec
-      , gdProtocolMagic    = pm
-      , gdAvvmDistr        = fakeAvvmDistr <> realAvvmMultiplied
+      { gdBootStakeholders   = bootStakeholders
```
Comment (dnadales) [inline src/Cardano/Chain/Genesis/Generate.hs:217]: Here's an example in which aligning is a bit annoying for reviewing purposes. Actually nothing is changed here besides the space. Just a side note. Sorry for the noise.

**Review by dnadales — APPROVED** [review]
:+1:

File: src/Cardano/Chain/Genesis/Generate.hs (hunk 2 above)
Comment (ruhatch) [reply src/Cardano/Chain/Genesis/Generate.hs:217]: No worries, I believe if you review with no whitespace changes you won't see these lines. That way we get nice alignment for readers and a good time for reviewers. I think you can set no whitespace as a default as well somehow.

**Pull request merged**